src/components/MultiHopTrade/components/TradeAmountInput.tsx

```diff
@@ -26,16 +27,30 @@ import { Balance } from 'components/DeFi/components/Balance'
 import { PercentOptionsButtonGroup } from 'components/DeFi/components/PercentOptionsButtonGroup'
 import { useLocaleFormatter } from 'hooks/useLocaleFormatter/useLocaleFormatter'
 import { bnOrZero } from 'lib/bignumber/bignumber'
+import { selectMarketDataByAssetIdUserCurrency } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
 import { colors } from 'theme/colors'
 
 import { usePriceImpactColor } from '../hooks/usePriceImpactColor'
 
+export type TradeAmountInputFormValues = {
+  amountFieldInput: string
```

Comment:
`amountFieldInput` is the actual input, the two others are virtual fields, derived and only set with `setValue` onChange.